pgmax/fg/graph.py

```diff
@@ -888,22 +887,18 @@ def BP(
     factor_type_to_msgs_range = bp_state.fg_state.factor_type_to_msgs_range
     factor_type_to_potentials_range = bp_state.fg_state.factor_type_to_potentials_range
 
-    def run_bp(
+    def bp_arrays_init(
```

Comment:
Instead of `init`, a better name here might actually be `update_bp_arrays` (since we are literally updating `bp_arrays`). And this function should take an additional optional argument `bp_arrays`. When the additional argument is `None` we grab values from `bp_state`.